sdk/shared/src/main/scala/org/ergoplatform/sdk/ContractTemplate.scala

```diff
@@ -0,0 +1,476 @@
+      val expressionTreeWriter = SigmaSerializer.startWriter()
+      ValueSerializer.serialize(ct.expressionTree, expressionTreeWriter)
+
+      Json.obj(
+        "treeVersion" -> ct.treeVersion.asJson,
+        "name" -> Json.fromString(ct.name),
+        "description" -> Json.fromString(ct.description),
+        "constTypes" -> ct.constTypes.asJson,
+        "constValues" -> (
+          if (ct.constValues.isEmpty) Json.Null
+          else ct.constValues.get.indices.map(i => ct.constValues.get(i) match {
+            case Some(const) => DataJsonEncoder.encodeData(const, ct.constTypes(i))
+            case None => Json.Null
+          }).asJson),
+        "parameters" -> ct.parameters.asJson,
+        "expressionTree" -> expressionTreeWriter.toBytes.asJson
+      )
+    })
+
+    implicit val decoder: Decoder[ContractTemplate] = Decoder.instance({ implicit cursor =>
+      val constTypesResult = cursor.downField("constTypes").as[IndexedSeq[SType]]
+      val expressionTreeBytesResult = cursor.downField("expressionTree").as[Array[Byte]]
+      (constTypesResult, expressionTreeBytesResult) match {
+        case (Right(constTypes), Right(expressionTreeBytes)) =>
+          val constValuesOpt = {
+            val constValuesJson = cursor.downField("constValues").focus.get
+            if (constValuesJson != Json.Null) {
+              val jsonValues = constValuesJson.asArray.get
+              Some(jsonValues.indices.map(
+                i => if (jsonValues(i) == Json.Null) None
+                else Some(DataJsonEncoder.decodeData(jsonValues(i), constTypes(i)))))
+            } else {
+              None
+            }
+          }
+
+          // Populate synthetic constants in the constant store for deserialization of expression tree.
+          val r = SigmaSerializer.startReader(expressionTreeBytes)
+          val constants = constTypes.indices.map(i => {
+            val t = constTypes(i)
+            DeserializationSigmaBuilder.mkConstant(defaultOf(t), t)
+          })
+          constants.foreach(c => r.constantStore.put(c)(DeserializationSigmaBuilder))
+
+          for {
+            treeVersion <- cursor.downField("treeVersion").as[Option[Byte]]
+            name <- cursor.downField("name").as[String]
+            description <- cursor.downField("description").as[String]
+            parameters <- cursor.downField("parameters").as[IndexedSeq[Parameter]]
+          } yield new ContractTemplate(
+            treeVersion,
+            name,
+            description,
+            constTypes,
+            constValuesOpt,
+            parameters,
+            ValueSerializer.deserialize(r).toSigmaProp)
+        case _ => Left(DecodingFailure("Failed to decode contract template", cursor.history))
+      }
+    })
+  }
+
+  /** Synthetic default value for the type.
+    * Used for deserializing contract templates.
+    */
+  private def defaultOf[T <: SType](tpe: T): T#WrappedType = {
```

Comment:
There was this Zero typeclass which I removed. Looks like it can be brought back https://github.com/ScorexFoundation/sigmastate-interpreter/commit/a064a0f71f8a4f40e41496c64280bd7e7a63670c

Reply:
Note that not all cases were handled in the original class. So I added additional code to handle them.

Reply:
For testing this code path, I was wondering if it would be nice to randomly generate `ErgoTree` in tests. Mostly to make sure that the `Zero` typeclass is properly tested. Do we use a fuzzer somewhere else in the code? If yes, please share.

Reply:
No, we don't have fuzzer. ErgoTree is too complex for fuzzer probably.

Reply:
ok, I guess the template generator is sufficient.